File: examples/typescript/sdk-reference/error-handling/linear-retry-strategy.ts
import {
  withDurableExecution,
  createLinearRetryStrategy,
  StepConfig,
} from "@aws/durable-execution-sdk-js";

const retryStrategy = createLinearRetryStrategy({
  maxAttempts: 5,
  initialDelay: { seconds: 2 },
  increment: { seconds: 3 },
  maxDelay: { seconds: 30 },
});

const stepConfig: StepConfig<string> = { retryStrategy };

export const handler = withDurableExecution(async (event, context) => {
  return context.step(
    "call-external-api",
    async () => callExternalApi(),
    stepConfig,
  );
});

async function callExternalApi(): Promise<string> {
  return "ok";
}

File: examples/java/sdk-reference/error-handling/linear-retry-strategy.java
import java.time.Duration;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.StepContext;
import software.amazon.lambda.durable.config.StepConfig;
import software.amazon.lambda.durable.retry.JitterStrategy;
import software.amazon.lambda.durable.retry.RetryStrategies;

public class LinearRetryStrategyExample extends DurableHandler<Object, String> {
    @Override
    public String handleRequest(Object input, DurableContext context) {
        StepConfig config = StepConfig.builder()
            .retryStrategy(RetryStrategies.linearBackoff(
                5,                       // maxAttempts
                Duration.ofSeconds(2),   // initialDelay
                Duration.ofSeconds(30),  // maxDelay
                Duration.ofSeconds(3),   // increment
                JitterStrategy.FULL))    // jitter
            .build();

        return context.step("call-external-api", String.class,
            (StepContext ctx) -> "ok",
            config);
    }
}
